src/Library.php

```diff
@@ -79,7 +74,57 @@ public static function maybe_create_tables() {
 			"
 		);
 
-		update_option( 'wc_blocks_db_version', \Automattic\WooCommerce\Blocks\Package::get_version() );
+		if ( ! $exists ) {
+			return self::add_create_table_notice( $table_name );
+		}
+
+		// Update succeeded. This is only updated when successful and validated.
+		// $schema_version should be incremented when changes to schema are made within this method.
+		update_option( 'wc_blocks_db_schema_version', $schema_version );
+	}
+
+	/**
+	 * Create database table, if it doesn't already exist.
+	 *
+	 * Based on admin/install-helper.php maybe_create_table function.
+	 *
+	 * @param string $table_name Database table name.
+	 * @param string $create_sql Create database table SQL.
+	 * @return bool False on error, true if already exists or success.
+	 */
+	protected static function maybe_create_table( $table_name, $create_sql ) {
+		global $wpdb;
+
+		if ( in_array( $table_name, $wpdb->get_col( 'SHOW TABLES', 0 ), true ) ) {
+			return true;
+		}
+
+		$wpdb->hide_errors();
```

Comment:
This will hide errors for this and all subsequent queries in the same request. We probably only want to hide the errors for this query only right? The function returns the _current_ value of `$wpdb->show_errors` before changing it so we can use that to reset back after executing the query.